### ISSUE TITLE
SSH support

### ISSUE THREAD

**Issue by ppalaga** [apache/camel-quarkus#798]

**Comment by squakez** [comment]
Hello there. I am interested in contributing to camel quarkus extensions, can I pick this component or anybody already worked on it by any chance?

**Comment by ppalaga** [comment]
Thanks, that's highly welcome! 

This guide may help: https://camel.apache.org/camel-quarkus/latest/contributor-guide/create-new-extension.html

**Comment by ppalaga** [comment]
Any progress, @squakez ? I'd add an autogenerated JVM-only extension unless you have a test at least?

**Comment by squakez** [comment]
Not really as I am focusing on another issue on the mongodb extension, so it may take still some day before I can focus on that.

**Comment by ppalaga** [comment]
A JVM-only extension (without a proper test) is underway.

**Comment by squakez** [comment]
@ppalaga I've started working on the native mode, and for that I made an ntegration test that I guess is valid also for the JVM mode. Do you want me to commit them and then keep ahead with the native development in parallel?

**Comment by ppalaga** [comment]
> an integration test that I guess is valid also for the JVM mode

That's great and yes, the test would also be useful for the JVM mode. 

> Do you want me to commit them

That's tricky. Your current branch effectively declares SSH as being native ready (in catalog and on the web page), which it is not, because as I understand from what you said that test is still not passing in native mode yet?

In theory, we could commit the test under `extensions-jvm/ssh/integration-test`. But I do not think it is worth the effort.

From my PoV, there is no hurry to commit the test, because we just released 1.1.0 and 1.2.0 will happen in a couple of weeks. So you have enough time to finish the native support. Feel free to ask if you have specific questions.
